Invalid content type 'application/x-ms-dos-executable'

It's my first time trying to use Coder with the JetBrains Toolbox and I can't get it to work. I just encounter following error:

```
Plugin com.coder.toolbox > Failed to configure coder.example.com: com.coder.toolbox.cli.ex.ResponseException: Invalid content type 'application/x-ms-dos-executable' when downloading CLI from https://coder.example.com/bin/coder-windows-amd64.exe. Expected application/octet-stream.
	at com.coder.toolbox.cli.downloader.CoderDownloadService.downloadCli(CoderDownloadService.kt:56)
	at com.coder.toolbox.cli.downloader.CoderDownloadService$downloadCli$1.invokeSuspend(CoderDownloadService.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
```

I installed coder using the install script and the implemented web server serves HTTPS. There is no reverse proxy involved.

I temporarily got it to work downloading the file manually, but that doesn't work anymore and it just resulted in this error:
```
Plugin com.coder.toolbox > Exception from plugin com.coder.toolbox in [com.jetbrains.toolbox.gateway.PluginServiceLocator$myKoinModule$lambda$35$lambda$2$$inlined$CoroutineExceptionHandler$1@685c3e22, CoroutineName(Service locator for Plugin com.coder.toolbox), kotlinx.coroutines.slf4j.MDCContext@5909531b, StandaloneCoroutine{Cancelling}@7326059e, Dispatchers.Default]: java.lang.NoClassDefFoundError: com/coder/toolbox/util/PathExtensionsKt
	at com.coder.toolbox.store.CoderSettingsStore.binPath(CoderSettingsStore.kt:126)
	at com.coder.toolbox.cli.downloader.CoderDownloadService.<init>(CoderDownloadService.kt:38)
	at com.coder.toolbox.cli.CoderCLIManager.createDownloadService(CoderCLIManager.kt:164)
	at com.coder.toolbox.cli.CoderCLIManager.<init>(CoderCLIManager.kt:141)
	at com.coder.toolbox.cli.CoderCLIManager.<init>(CoderCLIManager.kt:133)
	at com.coder.toolbox.cli.CoderCLIManagerKt.ensureCLI(CoderCLIManager.kt:73)
	at com.coder.toolbox.views.ConnectStep$connect$3.invokeSuspend(ConnectStep.kt:88)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:829)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
Caused by: java.lang.ClassNotFoundException: com.coder.toolbox.util.PathExtensionsKt
	... 13 more
```

How can I get this to work?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@
 ### Fixed
 
 - NPE during error reporting
+- relaxed `Content-Type` checks while downloading the CLI
 
 ## 0.6.3 - 2025-08-25
 

--- a/src/main/kotlin/com/coder/toolbox/cli/downloader/CoderDownloadService.kt
+++ b/src/main/kotlin/com/coder/toolbox/cli/downloader/CoderDownloadService.kt
@@ -25,6 +25,18 @@ import java.util.zip.GZIPInputStream
 import kotlin.io.path.name
 import kotlin.io.path.notExists
 
+private val SUPPORTED_BIN_MIME_TYPES = listOf(
+    "application/octet-stream",
+    "application/exe",
+    "application/dos-exe",
+    "application/msdos-windows",
+    "application/x-exe",
+    "application/x-msdownload",
+    "application/x-winexe",
+    "application/x-msdos-program",
+    "application/x-msdos-executable",
+    "application/vnd.microsoft.portable-executable"
+)
 /**
  * Handles the download steps of Coder CLI
  */
@@ -52,7 +64,7 @@ class CoderDownloadService(
         return when (response.code()) {
             HTTP_OK -> {
                 val contentType = response.headers()["Content-Type"]?.lowercase()
-                if (contentType?.startsWith("application/octet-stream") != true) {
+                if (contentType !in SUPPORTED_BIN_MIME_TYPES) {
                     throw ResponseException(
                         "Invalid content type '$contentType' when downloading CLI from $remoteBinaryURL. Expected application/octet-stream.",
                         response.code()

